ATOMIC_SWAP.md

```diff
@@ -0,0 +1,259 @@
+## Atomic-swap process
+
+The process can be divided in 2 phases:
+1. Negotiation phase
+  - Participants will coordinate the content of the atomic-swap.
+1. Signing phase
+  - Participants will sign the proposed transaction and exchange the signatures.
+  - A participant will collect the signatures and push the signed transaction on the network.
+
+## Decoding the proposal string
+
+A proposal string will be created and passed among participants during the atomic-swap and it can be confusing since it's not human readable.
+That's why you should use the decode API to get the current state of the proposal.
+
+### POST /wallet/decode
+
+Check what is contained on a proposal string.
+
+```bash
+$ curl -X POST -H "X-Wallet-Id: 123" -H "Content-type: application/json" --data "{\"partial_tx\":\"<proposalString>\"}" http://localhost:8000/wallet/decode
+{
+  "success": true,
+  "tx": {
+    "inputs": [
+      {
+        "txId": "00c7797738d890d1517f637f2af079d8daa4449fdfd098d2d636bb0e30c809d1",
+        "index": 1
+      }
+    ],
+    "outputs": [
+      {
+        "address": "WXf4xPLBn7HUC7F1U2vY4J5zwpsDS12bT6",
+        "value": 1,
+        "tokenData": 0
+      }
+    ]
+  }
+}
+```
+
+## How to: negotiation phase
+
+This phase consists of participants creating a proposal which will be serialized to a string which will be passed among participants.
+Each participan will add the tokens they want to send and receive from the transaction, once the transaction is complete the participants can move to the signing phase.
+
+1. Use the `/wallet/atomic-swap/tx-proposal` endpoint to get the proposal string and send to the participants.
+1. Each participant should update the proposal string using the `/wallet/atomic-swap/tx-proposal` endpoint and add their tokens and what they expect to receive.
+    - proposal strings cannot be merged, only updated, so each participant should take turns updating and sending the updated proposal string.
+1. The endpoint will return `isComplete: true` when the proposal string is ready to be signed.
+    - The proposal string can still be updated, but it will only be ready to sign again if the funds for inputs and outputs match.
+
+### POST /wallet/atomic-swap/tx-proposal
+
+Create a transaction proposal with multiple inputs and outputs.
+inputs, outputs, send_tokens, receive_tokens are operations which will add inputs and outputs to the partial_tx, can use one or many operations at once.
+
+*Body*:
+- partial_tx
+    - The current proposal string.
+    - If not present, will create a new proposal.
+- change_address
+    - Use this address for change where needed (e.g. sending 10 HTR but the UTXO has 15, a change output will be generated)
+    - If not present, an address from the caller wallet will be used.
+- send_tokens
+  - Will add inputs with the amount of tokens specified, may add a change output if needed.
+  - An array of objects with:
+    - value
+    - token: If not present will default to 00 (HTR)
+- receive_tokens
```

Comment:
It's not really clear the different between `receive_tokens` and `outputs`. They have exactly the same parameters

Reply:
The only actual difference is that for `receive_tokens`, the address is optional and if not present will generate an address from the wallet.

Reply:
I was not exactly talking about the low level difference, it's more about the guide level. If you are explaining to the people who will use the headless wallet, when they should use `receive_tokens` and when they should use `outputs`?

I'm afraid we are creating more parameters and complicating even more the end user decision.